runtime-light/stdlib/array/array-functions.h

```diff
@@ -13,6 +13,27 @@
 #include "runtime-light/coroutine/task.h"
 #include "runtime-light/stdlib/math/random-functions.h"
 
+namespace array_functions_impl_ {
+
+template<class T, class F>
+task_t<array<T>> array_filter_impl(const array<T> &a, const F &pred) noexcept {
+  array<T> result(a.size());
+  for (const auto &it : a) {
+    bool condition{false};
```

Comment:
The default value of `bool` is `false`, so `bool condition{};` is enough. But it's up to you